### ISSUE TITLE
Undefined property and call to a member function get() on null

### ISSUE THREAD

**Issue by Albrin** [ZF-Commons/ZfcUser#638]
After updating to the latest version ZfcUser package, I get the following error and warning:

```
Deprecated: ServiceLocatorAwareInterface is deprecated and will be removed in version 3.0, along with the ServiceLocatorAwareInitializer. Please update your ZfcTwig\View\HelperPluginManager plugin manager factory to inject the parent service locator via the constructor. in vendor/zendframework/zend-mvc/src/Service/ServiceManagerConfig.php on line 175

Deprecated: ServiceManagerAwareInterface is deprecated and will be removed in version 3.0, along with the ServiceManagerAwareInitializer. Please update your class ZfcUser\Service\User to remove the implementation, and start injecting your dependencies via factory instead. in vendor/zendframework/zend-mvc/src/Service/ServiceManagerConfig.php on line 129

Notice: Undefined property: ZfcUser\Factory\Controller\UserControllerFactory::$serviceLocator in vendor/zf-commons/zfc-user/src/ZfcUser/Factory/Controller/UserControllerFactory.php on line 38

Fatal error: Uncaught Error: Call to a member function get() on null in vendor/zf-commons/zfc-user/src/ZfcUser/Factory/Controller/UserControllerFactory.php:38 Stack trace: #0 [internal function]: ZfcUser\Factory\Controller\UserControllerFactory->createService(Object(Zend\Mvc\Controller\ControllerManager), 'zfcuser', 'zfcuser') #1 vendor/zendframework/zend-servicemanager/src/ServiceManager.php(934): call_user_func(Array, Object(Zend\Mvc\Controller\ControllerManager), 'zfcuser', 'zfcuser') #2 vendor/zendframework/zend-servicemanager/src/AbstractPluginManager.php(330): Zend\ServiceManager\ServiceManager->createServiceViaCallback(Array, 'zfcuser', 'zfcuser') #3 vendor/zendframework/zend-servicemanager/src/AbstractPlugin in vendor/zf-commons/zfc-user/src/ZfcUser/Factory/Controller/UserControllerFactory.php on line 38
```

I installed the following package versions:
```
$ ./composer.phar show -i
You are using the deprecated option "installed". Only installed packages are shown by default now. The --all option can be used to show all packages.
...
bjyoungblood/bjy-authorize          1.4.0              Zend\Acl based firewall system for ZF2 dispatch protection
container-interop/container-interop 1.1.0              Promoting the interoperability of container objects (DIC, SL, etc.)
doctrine/annotations                v1.2.7             Docblock Annotations Parser
doctrine/cache                      v1.6.0             Caching library offering an object-oriented API for many cache backends
doctrine/collections                v1.3.0             Collections Abstraction library
doctrine/common                     v2.6.1             Common Library for Doctrine projects
doctrine/dbal                       v2.5.4             Database Abstraction Layer
doctrine/doctrine-module            0.10.1             Zend Framework 2 Module that provides Doctrine basic functionality required for ORM and ODM modules
doctrine/doctrine-orm-module        0.9.2              Zend Framework 2 Module that provides Doctrine ORM functionality
doctrine/inflector                  v1.1.0             Common String Manipulations with regard to casing and singular/plural rules.
doctrine/instantiator               1.0.5              A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                      v1.0.1             Base library for a lexer that can be used in Top-Down, Recursive Descent Parsers.
doctrine/orm                        v2.5.4             Object-Relational-Mapper for PHP
dompdf/dompdf                       v0.6.2             DOMPDF is a CSS 2.1 compliant HTML to PDF converter
mpdf/mpdf                           v6.1.1             A PHP class to generate PDF files from HTML with Unicode/UTF-8 and CJK support
myclabs/deep-copy                   1.5.1              Create deep copies (clones) of your objects
phenx/php-font-lib                  0.2.2              A library to read, parse, export and make subsets of different types of font files.
phpdocumentor/reflection-docblock   2.0.4
phpoffice/phpexcel                  1.8.1              PHPExcel - OpenXML - Read, Create and Write Spreadsheet documents in PHP - Spreadsheet engine
phpspec/prophecy                    v1.6.0             Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage           3.3.1              Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator           1.4.1              FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template           1.2.1              Simple template engine.
phpunit/php-timer                   1.0.7              Utility class for timing
phpunit/php-token-stream            1.4.8              Wrapper around PHP's tokenizer extension.
phpunit/phpunit                     5.3.2              The PHP Unit Testing framework.
phpunit/phpunit-mock-objects        3.1.3              Mock Object library for PHPUnit
psr/http-message                    1.0                Common interface for HTTP messages
psr/log                             1.0.0              Common interface for logging libraries
sebastian/code-unit-reverse-lookup  1.0.0              Looks up which function or method a line of code belongs to
sebastian/comparator                1.2.0              Provides the functionality to compare PHP values for equality
sebastian/diff                      1.4.1              Diff implementation
sebastian/environment               1.3.6              Provides functionality to handle HHVM/PHP environments
sebastian/exporter                  1.2.1              Provides the functionality to export PHP variables for visualization
sebastian/global-state              1.1.1              Snapshotting of global state
sebastian/object-enumerator         1.0.0              Traverses array structures and object graphs to enumerate all referenced objects
sebastian/recursion-context         1.0.2              Provides functionality to recursively process PHP variables
sebastian/resource-operations       1.0.0              Provides a list of PHP built-in functions that operate on resources
sebastian/version                   1.0.6              Library that helps with managing the version number of Git-hosted PHP projects
setasign/fpdi                       1.6.1              FPDI is a collection of PHP classes facilitating developers to read pages from existing PDF documents and use them as templates in FPDF. Because it is also possible to use FPDI ...
symfony/console                     v3.0.5             Symfony Console Component
symfony/polyfill-mbstring           v1.1.1             Symfony polyfill for the Mbstring extension
symfony/yaml                        v3.0.5             Symfony Yaml Component
twig/twig                           v1.24.0            Twig, the flexible, fast, and secure template language for PHP
zendframework/zend-authentication   2.5.3              provides an API for authentication and includes concrete authentication adapters for common use case scenarios
zendframework/zend-barcode          2.6.0              provides a generic way to generate barcodes
zendframework/zend-cache            2.7.0              provides a generic way to cache any data
zendframework/zend-captcha          2.5.4
zendframework/zend-code             2.6.3              provides facilities to generate arbitrary code using an object oriented interface
zendframework/zend-config           2.6.0              provides a nested object property based user interface for accessing this configuration data within application code
zendframework/zend-console          2.6.0
zendframework/zend-crypt            2.6.0
zendframework/zend-db               2.8.1
zendframework/zend-debug            2.5.1
zendframework/zend-di               2.6.1
zendframework/zend-diactoros        1.3.5              PSR HTTP Message implementations
zendframework/zend-dom              2.6.0              provides tools for working with DOM documents and structures
zendframework/zend-escaper          2.5.1
zendframework/zend-eventmanager     2.6.3
zendframework/zend-feed             2.7.0              provides functionality for consuming RSS and Atom feeds
zendframework/zend-file             2.7.0
zendframework/zend-filter           2.7.1              provides a set of commonly needed data filters
zendframework/zend-form             2.8.3
zendframework/zend-http             2.5.4              provides an easy interface for performing Hyper-Text Transfer Protocol (HTTP) requests
zendframework/zend-hydrator         1.1.0
zendframework/zend-i18n             2.7.2
zendframework/zend-i18n-resources   2.5.2              Provides validator translations for zend-i18n's Translator
zendframework/zend-inputfilter      2.7.1
zendframework/zend-json             2.6.1              provides convenience methods for serializing native PHP to JSON and decoding JSON to native PHP
zendframework/zend-loader           2.5.1
zendframework/zend-log              2.8.2              component for general purpose logging
zendframework/zend-mail             2.7.0              provides generalized functionality to compose and send both text and MIME-compliant multipart e-mail messages
zendframework/zend-math             2.7.0
zendframework/zend-memory           2.5.1
zendframework/zend-mime             2.6.0
zendframework/zend-modulemanager    2.7.1
zendframework/zend-mvc              2.7.7
zendframework/zend-navigation       2.7.1              provides support for managing trees of pointers to web pages
zendframework/zend-paginator        2.7.0
zendframework/zend-permissions-acl  2.6.0              provides a lightweight and flexible access control list (ACL) implementation for privileges management
zendframework/zend-permissions-rbac 2.5.1              provides a role-based access control management
zendframework/zend-progressbar      2.5.2              component to create and update progressbars in different environments
zendframework/zend-psr7bridge       0.2.1              PSR-7 <-> Zend\Http bridge
zendframework/zend-serializer       2.7.1              provides an adapter based interface to simply generate storable representation of PHP types by different facilities, and recover
zendframework/zend-server           2.6.1
zendframework/zend-servicemanager   2.7.6
zendframework/zend-session          2.7.0              manage and preserve session data, a logical complement of cookie data, across multiple page requests by the same client
zendframework/zend-soap             2.6.0
zendframework/zend-stdlib           2.7.7
zendframework/zend-tag              2.6.1              a component suite which provides a facility to work with taggable Items
zendframework/zend-test             2.6.1
zendframework/zend-text             2.6.0
zendframework/zend-uri              2.5.2              a component that aids in manipulating and validating » Uniform Resource Identifiers (URIs)
zendframework/zend-validator        2.7.2              provides a set of commonly needed validators
zendframework/zend-version          2.5.1
zendframework/zend-view             2.6.7              provides a system of helpers, output filters, and variable escaping
zendframework/zend-xmlrpc           2.5.2
zendframework/zendframework         2.5.3              Zend Framework 2
zendframework/zendxml               1.0.2              Utility library for XML usage, best practices, and security in PHP
zf-commons/zfc-base                 v0.1.2             A set of genetic (abstract) classes which are commonly used across multiple modules.
zf-commons/zfc-twig                 1.2.2              Zend Framework 2 Module that provides a Twig rendering strategy and extensions to render actions or trigger events from your templates
zf-commons/zfc-user                 1.4.3              A generic user registration and authentication module for ZF2. Supports Zend\Db and Doctrine2.
zf-commons/zfc-user-doctrine-orm    1.0.1              Doctrine2 ORM storage adapter for ZfcUser.
```

**Comment by Danielss89** [comment]
Should be fixed in newest release(1.4.4 :)

**Comment by Albrin** [comment]
Thank you! After upgrading to 1.4.4 warning and error no.
